### ISSUE TITLE
[Linux] Option to treat all resource files as lowercase

### ISSUE THREAD

**Issue by incognico** [wootguy/resguy#1]
As my server runs Linux I've converted all resource file names (the actual resources like models/foo/bar.mdl) to lowercase, also in the curent .res files. The game client (at least on Windows, I don't know about Linux) deals with that just fine.

Well resguy does not do that:

Missing file "sprites/trempler/Spark3.spr" referenced in:
        sc_propanic_part1.bsp --> "testtrain1" (env_spritetrain)

but sprites/trempler/spark3.spr exists and is also in the current .res file.

This goes for all types of resources, for example in sprites/aom_weapons/weapon_aomglock.txt the sprite name is mixed case but the acual sprite on disk is in lowercase of course. Folder AomDC vs. aomdc.

My practice is probably suboptimal now that I think about the Linux client, will that one play a sound referenced in the map as sound/FOO.wav when the .res file on the server said sound/foo.wav and the game client downloaded that one? Can't verify this.

So basically I request a flag to treat all names of the gathered resourced as lowercase, this should not be that complicated to implement.

A better but more complicated way would be to somehow ignore the case sens on Linux though this requires more magic, like writing a function which which tests for every possible case combination of every resource and then test each one if it exists on the disk. sound/bAr/Fo_O1.wAv -> sound/[Bb][Aa][Rr]/[Ff][Oo]_[Oo]1\.[Ww][Aa][Vv]

Whatever, I'd be happy with a flag which just looks for the all-lowercase version on the disk.

**Comment by wootguy** [comment]
Personally, I think the map packages should be updated to use the correct case instead. Lowercasing every file on the server could mean people get duplicate (and possibly different) files when joining, which could cause problems. Or not, I haven't tested anything yet.

Anyway, try out the new -icase option.

**Comment by incognico** [comment]
```
[ 87%] Building CXX object CMakeFiles/resguy.dir/src/util.cpp.o
/home/srcds/resguy/src/util.cpp: In function ‘std::vector<std::__cxx11::basic_string<char> > get_script_dependencies(std::__cxx11::string)’:
/home/srcds/resguy/src/util.cpp:531:57: error: invalid initialization of non-const reference of type ‘std::__cxx11::string& {aka std::__cxx11::basic_string<char>&}’ from an rvalue of type ‘std::__cxx11::basic_string<char>’
      if (!contentExists(val) && !contentExists("sound/" + val))
                                                ~~~~~~~~~^~~~~
In file included from /home/srcds/resguy/src/util.cpp:1:0:
/home/srcds/resguy/src/util.h:114:6: note:   initializing argument 1 of ‘bool contentExists(std::__cxx11::string&, bool)’
 bool contentExists(string& file, bool fix_path=true);
      ^~~~~~~~~~~~~
CMakeFiles/resguy.dir/build.make:206: recipe for target 'CMakeFiles/resguy.dir/src/util.cpp.o' failed
make[2]: *** [CMakeFiles/resguy.dir/src/util.cpp.o] Error 1
CMakeFiles/Makefile2:67: recipe for target 'CMakeFiles/resguy.dir/all' failed
make[1]: *** [CMakeFiles/resguy.dir/all] Error 2
Makefile:83: recipe for target 'all' failed
make: *** [all] Error 2

**Comment by wootguy** [comment]
Oops. Fixed.

**Comment by incognico** [comment]
Cool, thank you! Works fine.

Probably unrelated: With io_v1.bsp I got a Failed to normalize path: ../svencoop/models/shell.mdl, the model is referenced in the weapon scripts but as "models/shell.mdl", seems okay to me, dunno why resguy says that. It then just puts models/shell.mdl into the .res even though it is in default_content.txt.

Edit: also for other maps:
```
Failed to normalize path: ../svencoop/models/shotgunshell.mdl
Failed to normalize path: ../svencoop/models/w_9mmarclip.mdl
Failed to normalize path: ../svencoop/models/mortarshell.mdl
Failed to normalize path: ../svencoop/sprites/red.spr
Failed to normalize path: ../svencoop/sprites/laserbeam.spr

**Comment by wootguy** [comment]
Looking into it. Suddenly the program is freezing for me when I get a normalize error in Linux.

**Comment by wootguy** [comment]
Ok, reproduced and fixed the normalize path errors. Seemed to only affect maps that use scripts. Try again? :>
